color-changer.js

```diff
@@ -0,0 +1,42 @@
+function getRandomUnusedHue(usedHues) {
+    let randomHue;
+    while(randomHue === undefined || usedHues[randomHue]) {
+        randomHue = Math.floor(Math.random()*360);
+    }
+    return randomHue;
+}
+
+function changeBgHue(newHue) {
+    let minLightness = 30;
+    let maxLightness = 70;
+    const randomLightness = Math.floor(Math.random() * (maxLightness - minLightness) + minLightness)
+    document.body.style.backgroundColor = `hsl(${newHue}, 50%, ${randomLightness}%)`;
```

Comment:
Absolutely—I'm a big fan of hsl for certain transition animations. Originally, I wanted a background color change that could transition to the next by going to the darkest version of the original color, changing the color, and then transitioning to about a mid-level of lightness for it.

I opted for a much simpler transition method, but if I wanted to pursue that route, I'd just have to adjust one variable (lightness) instead of doing some RGB or HEX calculations.

Plus, once you know the 360 degree hue color wheel, the concept is just easiest for me to grok through. Good readability